src/components/Form.js

```diff
@@ -1,22 +1,41 @@
-import React from 'react'
+import React, { useState } from 'react'
 import { Button } from './Button'
 import { TextInput } from './TextInput'
 import styles from './Form.module.css'
+import { useFormik } from 'formik'
+import { object, string } from 'yup'
+import { Link } from 'react-router-dom'
 
 export const Form = ({ type = 'signup' }) => {
+  const userSchema = object({
+    username: string().min(3),
+    password: string().min(6)
+  })
+
+  const formik = useFormik({
+    initialValues: { username: '', password: '' },
+    validationSchema: userSchema,
+    onSubmit: (values) => {
+      // log the values of the form
+      console.log('Password: ' + values.password)
+      console.log('Username: ' + values.username)
+    }
+  })
   let submitButton
   let linkToForm
   if (type === 'signin') {
-    submitButton = <Button>Sign In</Button>
-    linkToForm = <a href="#">Sign Up</a>
+    submitButton = <Button onClick="Sign In Form ">Sign In</Button>
+    linkToForm = <Link to='/sign-up'>Sign Up</Link>
   } else {
-    submitButton = <Button>Sign Up</Button>
-    linkToForm = <a href="#">Sign In</a>
+    linkToForm = <Link to='/sign-in'>Sign In</Link>
+    submitButton = <Button onClick="Sign Up Form">Sign Up</Button>
   }
   return (
-    <form >
-      <TextInput type="text" id="username" label="Username"></TextInput>
-      <TextInput type="password" id="password" label="password"></TextInput>
+    <form onSubmit={formik.handleSubmit}>
+      <TextInput type="text" id="username" name="username" label="Username" onChange={formik.handleChange} value={formik.values.username} ></TextInput>
+      { formik.errors.username }
+      <TextInput type="password" id="password" name="password" label="password" onChange={formik.handleChange} value={formik.values.password}></TextInput>
```

Comment:
I think I would try to add the error messages to the `TextInput` / `DecimalInput` components.